management-api-common/src/main/java/com/datastax/mgmtapi/shims/CassandraAPI.java

```diff
@@ -30,6 +30,12 @@
 public interface CassandraAPI
 {
     void decommission(boolean force) throws InterruptedException;
+    
+    void enableFullQuerylog() throws UnsupportedOperationException;
+
+    void disableFullQuerylog()throws UnsupportedOperationException;
+    
+    boolean isFullQueryLogEnabled() throws UnsupportedOperationException;
```

Comment:
So for these 3 new methods, I think a better approach would be to provide a default implementation in the interface here, and only override them in CassandraAPI4x.
```suggestion
    default void enableFullQuerylog() throws UnsupportedOperationException
    {
        throw new UnsupportedOperationException("FQL is only supported on OSS Cassandra > 4x");
    };

    default void disableFullQuerylog()throws UnsupportedOperationException;
    {
        throw new UnsupportedOperationException("FQL is only supported on OSS Cassandra > 4x");
    };

    default boolean isFullQueryLogEnabled() throws UnsupportedOperationException;
    {
        throw new UnsupportedOperationException("FQL is only supported on OSS Cassandra > 4x");
    };
```

With the above, you can remove the overrides in the CassandraAPI3x class and the DseAPI68 class.

Reply:
I like the direction of this thought... So the idea is we default to `UnsupportedOperationException("FQL is only supported on OSS Cassandra > 4x");` and override for 4x. 

My only question: we may want to switch this around so that the override is on the CassandraAPI3x class and the DseAPI68 class. This is because the feature will exist in hypothetical future CassandraAPI5x classes etc. so eventually it is the error case that will be the exception.

Reply:
Oh also, these will be the only methods with default implementations in that interface. Just noting in case we want to reconsider for consistency.

Reply:
Yes, the `default` mechanism here is meant to provide the most basic functionality across all implementations and can't access anything not at the interface level. Since we have 2 agents that will just thrown an unsupported Exception (3.11 and DSE 6.8), that's about all we can do here.
Going forward, a Cassandra 5.x agent might need to do something different than the 4.x code, so there's no guarantees that a common implementation would exist between them (however unlikely that is in this specific case).